File: code/native/swift/runtime/math.swift
import Foundation

enum math {
    static func abs(_ v: Int) -> Int { return Swift.abs(v) }
    static func min(_ a: Int, _ b: Int) -> Int { return Swift.min(a, b) }
    static func max(_ a: Int, _ b: Int) -> Int { return Swift.max(a, b) }
    static func pow(_ base: Int, _ exponent: Int) -> Int { return Int(Foundation.pow(Double(base), Double(exponent))) }
    static func sign(_ v: Int) -> Int { return v > 0 ? 1 : (v < 0 ? -1 : 0) }
    static func sqrt(_ v: Int) -> Int { return Int(Double(v).squareRoot()) }
    static func floor(_ v: Int) -> Int { return v }
    static func ceil(_ v: Int) -> Int { return v }
    static func round(_ v: Int) -> Int { return v }
}

File: code/native/rust/runtime/math.rs
mod math {
    pub fn abs(v: i64) -> i64 { v.abs() }
    pub fn min(a: i64, b: i64) -> i64 { a.min(b) }
    pub fn max(a: i64, b: i64) -> i64 { a.max(b) }
    pub fn pow(base: i64, exponent: i64) -> i64 { base.pow(exponent as u32) }
    pub fn sign(v: i64) -> i64 { v.signum() }
    pub fn sqrt(v: i64) -> i64 { (v as f64).sqrt() as i64 }
    pub fn floor(v: i64) -> i64 { v }
    pub fn ceil(v: i64) -> i64 { v }
    pub fn round(v: i64) -> i64 { v }
}
